pkg/controller/statusmanager/pod_status.go

```diff
@@ -72,31 +73,49 @@ func FindStatusCondition(conditions []corev1.NodeCondition, conditionType corev1
 	return nil
 }
 
-func SetNodeCondition(conditions *[]corev1.NodeCondition, newCondition corev1.NodeCondition) {
+func SetNodeCondition(conditions *[]corev1.NodeCondition, newCondition corev1.NodeCondition) (changed bool) {
+	changed = false
 	if conditions == nil {
 		conditions = &[]corev1.NodeCondition{}
 	}
 	existingCondition := FindStatusCondition(*conditions, newCondition.Type)
 	if existingCondition == nil {
 		newCondition.LastTransitionTime = metav1.NewTime(time.Now())
 		*conditions = append(*conditions, newCondition)
-		return
+		return true
 	}
 
 	if existingCondition.Status != newCondition.Status {
 		existingCondition.Status = newCondition.Status
 		existingCondition.LastTransitionTime = metav1.NewTime(time.Now())
+		changed = true
 	}
 
 	existingCondition.Reason = newCondition.Reason
 	existingCondition.Message = newCondition.Message
+	return changed
+}
+
+func canRemoveNetworkUnavailable(sharedInfo *sharedinfo.SharedInfo) bool {
+	now := time.Now()
+	if sharedInfo.LastNetworkAvailabe == nil {
+		sharedInfo.LastNetworkAvailabe = &now
+		return false
+	}
+	log.Info(fmt.Sprintf("Last network unavailable time is %s", *sharedInfo.LastNetworkAvailabe))
+	// A Duration represents the elapsed time between two instants as an int64 nanosecond count
+	if int64(now.Sub(*sharedInfo.LastNetworkAvailabe)) < operatortypes.TimeBeforeRecoverNetwork * 1000000000 {
```

Comment:
Is it possible to add a timer to update status?

Reply:
done